test/ui/cypress/CYPRESS.md

```diff
@@ -0,0 +1,72 @@
+# Cypress Testing
+
+Cypress is being used for both end-to-end tests and component tests.
+
+## Prerequisites
+
+1. You have an AAP instance being deployed on Azure.
+2. The Deployment Engine UI is accessible.
+
+## Run
+
+1. Initialize your local environment.
+   You may need to create a virtualenv and install the needed packages from ui/requirements.txt if you don't already have them.
+
+    ```shell
+    cd <repo>/test/ui
+    python -m venv venv
+    source venv/bin/activate    
+
+    # Clean install a project
+    npm ci
+
+    # Verify Cypress installation
+    npx cypress verify
+
+    # Install the following dependencies to see useful messages when the engine is dead.
+    # This is optional, not manadatory.
+    cd <repo>/ui
+    pip install -r requirements.txt
+    ```
+
+2. Configure tests input.
+
+   Option 1: Set environment variables at OS level
+   ```shell
+   export CYPRESS_baseUrl=<Deployment Engine UI Url>
+   export CYPRESS_DEPLOYMENT_ENGINE_UI_PASSWORD=<Admin password to login Deployment Engine UI>
+   export CYPRESS_RH_SSO_URL=https://sso.redhat.com
+   export CYPRESS_RH_ACCOUNT_USERNAME=<User to login https://sso.redhat.com>
+   export CYPRESS_RH_ACCOUNT_PASSWORD=<Password to login https://sso.redhat.com>
+   ```
+   Option 2: Set environment variables in ui/cypress.config.ts
```

Comment:
Changing the config file will create a change in a file that's tracked in git so users will need to undo that change if they want to pull in fresh code. Instead, we should use `cypress.env.json` and have it in git ignore file.

Reply:
Agree. I will add this file into the `.gitignore' file and leave an example in the `CYPRESS.md` file.

Reply:
The example should only talk about adding the `cypress.env.json`, not changing the `cypress.config.js`.

Reply:
I am afraid if we define cypress env variable configuration in both `cypress.env.json` and `cypress.config.js`, it will bring some confusion. I made some change for it to mention `cypress.env.json` only.